python/ray/dashboard/res/main.js

```diff
@@ -0,0 +1,263 @@
+let dashboard = new Vue({
+    el: "#dashboard",
+    data: {
+        now: (new Date()).getTime() / 1000,
+        shown: {},
+        error: "loading...",
+        last_update: undefined,
+        clients: undefined,
+        totals: undefined,
+        tasks: undefined,
+        ray_config: undefined,
+    },
+    methods: {
+        updateNodeInfo() {
+            axios.get("/api/node_info").then(function (resp) {
+                setTimeout(dashboard.updateNodeInfo, 500);
+
+                dashboard.error = resp.data.error;
+                dashboard.last_update = resp.data.timestamp;
+                if (resp.data.error) {
+                    dashboard.clients = undefined;
+                    dashboard.tasks = undefined;
+                    dashboard.totals = undefined;
+                    return;
+                }
+                dashboard.clients = resp.data.result.clients;
+                dashboard.tasks = resp.data.result.tasks;
+                dashboard.totals = resp.data.result.totals;
+            }).catch(function() {
+                setTimeout(dashboard.updateNodeInfo, 500);
```

Comment:
This seems to be called after the then(). When that happens you end up with an exponentially increasing number of requests.

Reply:
I think the solution to this is putting the update call in a .finally() block